src/include/storage/storage_structure/disk_array.h

```diff
@@ -137,7 +139,113 @@ class BaseDiskArrayInternal {
 
     virtual void prepareCommit();
 
+    // Write Iterator for making fast bulk changes to the disk array
+    // The pages are cached while the elements are stored on the same page
+    // Designed for sequential writes, but supports random writes too (at the cost that the page
+    // caching is only beneficial when seeking from one element to another on the same page)
+    struct Iterator {
+        BaseDiskArrayInternal& diskArray;
+        PageCursor apCursor;
+        uint32_t valueSize;
+        // TODO(bmwinger): Instead of pinning the page and updating in-place, it might be better to
+        // read and cache the page, then write the page to the WAL if it's ever modified. However
+        // when doing bulk hashindex inserts, there's a high likelihood that every page accessed
+        // will be modified, so it may be faster this way.
+        WALPageIdxAndFrame walPageIdxAndFrame;
+        static const transaction::TransactionType TRX_TYPE = transaction::TransactionType::WRITE;
+        uint64_t idx;
+        uint64_t originalNumElements;
+        std::unique_lock<std::shared_mutex> lock;
+        DEFAULT_MOVE_CONSTRUCT(Iterator);
+
+        // Constructs Iterator in an invalid state. Seek must be called before accessing data
+        Iterator(uint32_t valueSize, BaseDiskArrayInternal& diskArray,
```

Comment:
`BatchWriteIterator` looks a better name to me.

Reply:
How about just WriteIterator? The fact that it's an iterator implies you'd usually use it for batch updates, though I doubt it would be noticeably slower than `BaseDiskArray::update` for single updates.

Reply:
Sure. `WriteIterator` is clear.